src/pages/Validators/ValidatorsMap.tsx

```diff
@@ -1,38 +1,71 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 import {Stack, useMediaQuery, useTheme} from "@mui/material";
 import Map from "./Components/Map";
 import {grey} from "../../themes/colors/aptosColorPalette";
 import MapMetrics from "./Components/MapMetrics";
 import {useGetValidatorSetGeoData} from "../../api/hooks/useGetValidatorSetGeoData";
+import {useGetEpochTime} from "../../api/hooks/useGetEpochTime";
+import {useGetValidatorSet} from "../../api/hooks/useGetValidatorSet";
+import {useGetStakingRewardsRate} from "../../api/hooks/useGetStakingRewardsRate";
+import {SkeletonTheme} from "react-loading-skeleton";
 
 export default function ValidatorsMap() {
   const theme = useTheme();
+  const isDarkTheme = theme.palette.mode === "dark";
   const isOnMobile = !useMediaQuery(theme.breakpoints.up("md"));
-  const backgroundColor = theme.palette.mode === "dark" ? grey[800] : grey[50];
+  const backgroundColor = isDarkTheme ? grey[800] : grey[50];
 
   const {validatorGeoMetric, validatorGeoGroups} = useGetValidatorSetGeoData();
+  const [isSkeletonLoading, setIsSkeletonLoading] = useState<boolean>(true);
+  const {curEpoch} = useGetEpochTime();
+  const {totalVotingPower} = useGetValidatorSet();
+  const {rewardsRateYearly} = useGetStakingRewardsRate();
+  const {numberOfActiveValidators} = useGetValidatorSet();
 
-  return isOnMobile ? (
-    <Stack
-      direction="column"
-      justifyContent="space-between"
-      marginY={4}
-      sx={{backgroundColor: backgroundColor}}
-      overflow="hidden"
-    >
-      <Map validatorGeoGroups={validatorGeoGroups} />
-      <MapMetrics validatorGeoMetric={validatorGeoMetric} isOnMobile />
-    </Stack>
-  ) : (
-    <Stack
-      direction="row"
-      justifyContent="space-between"
-      marginY={4}
-      sx={{backgroundColor: backgroundColor}}
-      overflow="hidden"
-    >
-      <MapMetrics validatorGeoMetric={validatorGeoMetric} />
-      <Map validatorGeoGroups={validatorGeoGroups} />
-    </Stack>
+  useEffect(() => {
+    if (
+      curEpoch &&
+      totalVotingPower &&
+      rewardsRateYearly &&
+      numberOfActiveValidators
+    ) {
+      setIsSkeletonLoading(false);
+    }
+  }, [curEpoch, totalVotingPower, rewardsRateYearly, numberOfActiveValidators]);
+
+  return (
+    <SkeletonTheme baseColor={isDarkTheme ? grey[500] : undefined}>
```

Comment:
what's `baseColor` here?

Reply:
this's the skeleton theme's loading color, which should match the theme we have either light or dark.